diff --git a/node_modules/nativescript-vue/dist/index.js b/node_modules/nativescript-vue/dist/index.js
index fa6f225..74cb59a 100644
--- a/node_modules/nativescript-vue/dist/index.js
+++ b/node_modules/nativescript-vue/dist/index.js
@@ -50,6 +50,10 @@ export const createApp = ((...args) => {
         return mount(...args);
     };
     app.start = () => {
+        // Patch: expose app instance globally for HMR access
+        // This primarily allows back navigation to be restored
+        // when doing HMR within deep navigation stacks
+        globalThis.__NS_VUE_APP__ = app;
         const componentInstance = app.mount(createAppRoot(), false, false);
         startApp(componentInstance);
         setRootApp(app);
